src/main/java/org/testng/internal/BaseTestMethod.java

```diff
@@ -743,4 +766,20 @@ public IParameterInfo getFactoryMethodParamsInfo() {
     }
     return null;
   }
+
+  private IRetryAnalyzer getRetryAnalyzerConsideringMethodParameteters(ITestResult tr) {
+    Object[] key = tr.getParameters();

```

Comment:
My last problem is here because it is not really obvious why parameters are on the result and not on the method.

But if I remember well, TestMethod is something like the static description of the method (1 instance by method) and TestMethodInstance doesn't exist for the moment (1 instance by method run). So TestResult is the only way to catch parameters.

Reply:
True. Parameter info is better off if its bundled with a test method. But then again we need to dig in deeper to figure out how to refactor that. We can do that in a separate PR.